demisto_sdk/commands/content_graph/parsers/json_content_item.py

```diff
@@ -99,4 +99,7 @@ def get_json(self, git_sha: Optional[str]) -> Dict[str, Any]:
                     f"Directory {self.path} must have a single JSON file."
                 )
             self.path = Path(json_files_in_dir[0])
+        if not self.path.suffix.lower() == ".json":
+            raise NotAContentItemException
```

Comment:
As we changed the behavior of the `from_path` method, when parsing the content repository instead of raising a `NonContentItem` exception on non JSON/YAML files, the `by_path` finds their content type (from the path) and we are failing to parse them. That is why extra validation is needed.